content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-bitbucket-identity.md

```diff
@@ -0,0 +1,422 @@
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 4 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + bitbucket-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `bitbucket-identity` value. Note that you may receive a `PermissionDenied` error part way through the apply step. If so, run `chainctl auth login` once more, and then `terraform apply` again to resume creating the identity and resources.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+bitbucket-identity = "<your bitbucket identity>"
+```
+
+This is the identity's [UIDP (unique identity path)](/chainguard/chainguard-enforce/reference/events/#uidp-identifiers), which you configured the `bitbucket.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Bitbucket workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
+
+```sh
+chainctl iam identities ls
+```
+
+You're now ready to edit a Bitbucket pipeline in order to test out this identity.
+
+
+## Testing the identity with a Bitbucket pipeline
+
+To test the identity you created with Terraform in the previous section, ensure you have Pipelines enabled for your repository and then create a `bitbucket-pipelines.yml` file in the root of your repository. Note: if you already have a pipeline with steps defined then you only need to add the `oidc: true` field to your pipeline to enable OIDC for the step in question.
+
+Copy the following pipeline defintion into your `bitbucket-pipelines.yml` file and commit it to the repository.
+
+```
+image: atlassian/default-image:3
+
+pipelines:
+  default:
+    - step:
+        oidc: true
+        max-time: 5
+        script:
+          - curl -o chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m)"
+          - chmod +x chainctl
+
+          # Assume the bitbucket pipeline identity
+          - ./chainctl auth login --identity-token $BITBUCKET_STEP_OIDC_TOKEN --identity <bitbucket-identity>
+          - ./chainctl policy ls
+```
+
+The important line is the `oidc: true` option, which enables OIDC for the individual step in the pipeline. This configuration is why the `subject_pattern` is used in the Terraform configuration, since each step gets its own UUID identifier, which is added to the `sub` field in the generated OIDC token.
+
+Now you can add the commands for testing the identity like `chainctl policy ls` in the following example:
```

Comment:
This + the following code block feel redundant since you just showed these lines in the previous block